package.json

```diff
@@ -9,6 +9,8 @@
     "build": "tsc",
     "dev": "nodemon src/index.ts",
     "start": "node build/src/index.js",
+    "test:unit": "jest --runInBand",
+    "test:unit:watch": "jest --watch",
```

Comment:
Can you run the test with the `dotenv -e test/.env.test` prefix, so we boot them in the test env like the other tests ?

Reply:
why do we need it? For unit tests better to mock all functions as well as `process.env`

Reply:
the idea is to separate integration and unit tests. Unit tests could cover small pieces of code and give a quick response if something is broken. While integration/e2e tests are heavier and probably will not cover all possible scenarios.

Reply:
I have separated configs for e2e tests and unit tests

Reply:
So you have a separate config for E2E tests, only for faster tests ? And these units tests can still be ran using the default jest.config.ts ? (so `yarn test` can run all tests, e2e and units)

Reply:
About speed: the purpose of unit tests is to test units (functions, classes, files). So if in unit tests if I test some function, everything else is mocked. On the other hand, e2e tests could test the service in integration (DB connection, fetching data from external resources). That's why unit tests are faster and less dependent on third parties.

We shouldn't have the same config for e2e and unit tests.

Reply:
you are right. Separated.